gbrl/src/cpp/compression.cpp

```
@@ -0,0 +1,368 @@
+ * @param edata Ensemble data structure
+ * @param metadata Ensemble metadata
+ * @param start_tree_idx First tree to process (inclusive)
+ * @param stop_tree_idx Last tree to process (exclusive)
+ * @param matrix Output matrix representation
+ */
+void Compressor::get_representation_matrix_over_leaves(const float *obs, const char *categorical_obs, const int sample_idx, const ensembleData *edata, const ensembleMetaData *metadata, const int start_tree_idx, const int stop_tree_idx, matrixRepresentation *matrix){
+    const int max_depth = metadata->max_depth;
+    const int n_num_features = metadata->n_num_features;
+    const int n_cat_features = metadata->n_cat_features;
+    const int n_leaves = metadata->n_leaves;
+    const int n_leaves_plus_one = n_leaves + 1;
```

Cppcheck (2.19.0)

[style] The function 'valid_tree_idx' is never used.

(unusedFunction)